src/services/item/plugins/itemTag/repository.ts

```diff
@@ -225,6 +225,11 @@ export class ItemTagRepository {
    * @param  {Item[]} items
    */
   async getForManyItems(items: Item[], { withDeleted = false }: { withDeleted?: boolean } = {}) {
+    // should not query when items array is empty
+    if (!items.length) {
+      throw new Error('empty items list joins on full table');
+    }
+
```

Comment:
It is more a general question, but should we throw in the Repository when something is not found (or here because the items are empty) or should we return null (when single value) or an empty array ? What do you think @pyphilia @Defferrard ?

Reply:
I would have returned an empty result here?

Reply:
@pyphilia @ReidyT I want to throw an error, because I do not think that calling this function with an empty array will give the expected result. I would consider it a violation of the usage of the function. So given this I think it makes sens to throw instead of silently ignoring. I think if we silently ignore, it will make it harder to debug bad uses of this function, as it makes it harder to know iff you got nothing because there was nothing or because you supplied an empty array. 

I am however open to argumentation.

Reply:
I fell like it is a multiplication of 0, which results to 0. I don't know if the comparaison is explicit, but I'd not expect the function to crash if I give an empty array.
Having an exception for this case means that we would probably have to add a check before each call of this function. If we want to respect explicit exceptions, on this situation you'd add a `@throw` comment on the function's JSDoc, on all other functions that calls it and also on the AJV Schema.
Moreover, I don't think we should define our interface by how we need it to be debugged.

If we decide to keep the exception here, we'd have to throw a more explicit error instead of `new Error(...)`